src/components/button/styles.ts

```diff
@@ -0,0 +1,289 @@
+    &:hover {
+      background: var(--fill-button-primary-hover);
+      .${elButtonIcon} {
+        color: var(--icon-button_primary-hover);
+      }
+    }
+  }
+
+  &.${elButtonSecondary} {
+    &:hover {
+      color: var(--text-tertiary);
+      border: 1px solid var(--outline-button-secondary-hover);
+      .${elButtonIcon} {
+        color: var(--icon-secondary);
+      }
+    }
+  }
+
+  &.${elButtonTertiary} {
+    border: unset;
+    &:hover {
+      color: var(--text-tertiary);
+      .${elButtonIcon} {
+        color: var(--icon-secondary);
+      }
+    }
+  }
+
+  &.${elButtonDestructive} {
+    background: var(--fill-error-dark);
+    color: var(--text-white);
+    border: unset;
+    .${elButtonIcon} {
+      color: var(--icon-button_danger-default);
+    }
+    &:hover {
+      background: var(--fill-button-danger-hover);
+      .${elButtonIcon} {
+        color: var(--icon-button_danger-hover);
+      }
+    }
+  }
+
+  &[disabled],
+  &.${elButtonDisabled},
+  &.${elButtonBusy},
+  &[disabled].${elButtonPrimary},
+  &[disabled].${elButtonSecondary},
+  &[disabled].${elButtonTertiary},
+  &[disabled].${elButtonDestructive},
+  &.${elButtonBusy}
+    .${elButtonPrimary},
+  &.${elButtonBusy}
+    .${elButtonSecondary},
+  &.${elButtonBusy}
+    .${elButtonTertiary},
+  &.${elButtonBusy}
+    .${elButtonDestructive} {
+    color: var(--text-placeholder);
+    background-image: linear-gradient(to right, var(--neutral-100), var(--neutral-100));
+    cursor: not-allowed;
+    border: unset;
+
+    &:hover {
+      .${elButtonIcon} {
+        color: var(--text-placeholder);
+      }
+    }
+  }
+
+  &:focus {
+    box-shadow:
+      0 0 0 1px white,
+      0px 0px 0px 4px var(--blue-300);
+  }
+  
+  &.${elFloatingButton} {
+    border-radius: 100%;
+    height: 3.75rem;
+    width: 3.75rem;
+    margin: 0.5rem;
+    max-height: unset;
+    padding: 0.4rem;
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+
+    ${isMobile} {
+      width: 2.5rem;
+      height: 2.5rem;
+      margin: 0.5rem;
+    }
+
+    ${elButtonIcon} {
+      color: var(--neutral-400);
+
+      &:hover {
+        color: var(--neutral-700);
+      }
+    }
+
+    &.${elButtonDestructive}, &.${elButtonPrimary} {
+      > .${elButtonIcon} {
+        color: var(--white);
+      }
+    }
+
+    &.${elButtonBusy} {
+      & ${elButtonSpinner} {
+        left: inherit;
+        top: inherit;
+      }
+
+      > .${elButtonIcon} {
+        visibility: hidden;
+      }
+    }
+  }
+`
+
+// Apply shared styles to button component
+export const ElButton = styled.button`
+  ${baseButtonStyles}
+  &.${elButtonBusy} {
+    ${ElButtonSpinner} {
+      display: block;
+    }
+  }
+`
+
+// Apply shared styles to anchor component
+export const ElAnchorButton = styled.a`
+  ${baseButtonStyles}
+  &.${elButtonBusy} {
+    ${ElButtonSpinner} {
+      display: block;
+    }
+  }
+`
+
+// Button icon styling
+export const ElButtonIcon = styled(Icon)`
```

Comment:
I was not aware of this decision and was under the impression that only Elements icons need to be supported as v4. _(Let's discuss this)_

Reply:
@kurtdoherty I have confirmed with Andrei, that all icons should come from the elements library. But we can discuss this in our meeting next week.

Reply:
I agree that only Design System icons should be used in Reapit products, but I think in practice, we need to allow product engineering teams the flexibility to get ahead of Elements. Also, I think we want the flexibility within Elements itself to change how we handle icons without having to be concerned about breaking other components that have tightly coupled themselves to the existing interface.

This effectively boils down to the _"interface segregation principle"_. Button doesn't need to be concerned with the interface of an Icon. It just needs to all an icon to be rendered on the left or right side of the button's label. The exact interface a consumer uses to render an icon is best handled by the Icon atom.

To explore this further, here's two likely scenarios we'll face in future:

**A new product feature needs a new icon that isn't supported by Elements**

With the current interface, they can't render the new icon SVGs in their product on their own timeline; rather, they must:
- Make a contribution to Elements;
- Have it reviewed by both Element's devs and the Design team;
- Have a new Elements release published;
- Update their product to that new version (at worst, dealing with any breaking changes that upgrade may incur);
- Use the newly available icon/s.

However, if the button's icon props simply accept a ReactNode, this scenario boils down to:
- Supply a custom SVG as the icon and release their feature;
- Create an issue in Elements highlighting the need for the new icon/s;
- When Elements is updated with support for the new icons, and when the product engineering team is ready to upgrade Elements:
  - Update their product to that new version (at worst, dealing with any breaking changes that upgrade may incur); 
  - Use the newly available icon/s instead of custom SVG.

**The Elements team decides to overhaul the icon implementation (e.g. to facilitate a dual tone colour schema, or two optimise consumer bundle sizes, or ...):**

If we want to replace `Icon`, and we have many Elements components that are tightly coupled to Icon's API (e.g. an icon name), we'd have to:
- Prepare the replacement Icon component;
- Update all uses of the old icon component in all other Elements components;
- If we DIDN'T care about consumer migration journey's (_unlikely_):
  - Release a new version of Elements;
  - Ask all product engineering teams to upgrade and deal with the breaking changes in the interfaces of all updated Elements components;
  - Reap the benefits of the new API in any product that does the upgrade (at the cost of making the upgrade more costly).
- If we DID care about consumer migration journey's:
  -  Ensure all the changes are backwards compatible with the old icon component's interface, and the interfaces of the other Elements components that are coupled to it (which could severely limit the benefits the icon component is trying to achieve)
  - Deprecate the backwards compatible API and ask product engineering teams to migrate away from it;
  - Finally reap the benefits of the new API.

However, if the button's icon props simply accept a ReactNode, this scenario boils down to:
- Release the new icon component and deprecate the old one;
- Announce to product engineering teams that the latest version of Elements has a new icon component and explain why they should use it;
- Allow product engineering teams to swap out `Icon` for its replacement.